provider/docker.go

```diff
@@ -621,6 +622,48 @@ func (provider *Docker) getPassHostHeader(container dockerData) string {
 	return "true"
 }
 
+func (provider *Docker) getCORS(container dockerData) *types.CORS {
+	hasCORS := false
+	c := &types.CORS{}
+	if value, err := getLabel(container, "traefik.frontend.cors.allowedOrigins"); err == nil {
+		c.AllowedOrigins = trimSpaceArray(strings.Split(value, ","))
+		hasCORS = true
+	}
+	if value, err := getLabel(container, "traefik.frontend.cors.allowedMethods"); err == nil {
+		c.AllowedMethods = trimSpaceArray(strings.Split(value, ","))
+		hasCORS = true
+	}
+	if value, err := getLabel(container, "traefik.frontend.cors.allowedHeaders"); err == nil {
+		c.AllowedHeaders = trimSpaceArray(strings.Split(value, ","))
+		hasCORS = true
+	}
+	if value, err := getLabel(container, "traefik.frontend.cors.exposedHeaders"); err == nil {
+		c.ExposedHeaders = trimSpaceArray(strings.Split(value, ","))
+		hasCORS = true
+	}
+	if value, err := getLabel(container, "traefik.frontend.cors.allowCredentials"); err == nil {
+		c.AllowCredentials = value == "true"
+		hasCORS = true
+	}
+	if value, err := getLabel(container, "traefik.frontend.cors.maxAge"); err == nil {
+		maxAge, errConv := strconv.Atoi(value)
+		if errConv != nil {
+			log.Errorf("Unable to parse traefik.frontend.cors.maxAge %s", value)
+		} else {
+			c.MaxAge = maxAge
+			hasCORS = true
+		}
+	}
+	if value, err := getLabel(container, "traefik.frontend.cors.optionsPassthrough"); err == nil {
+		c.OptionsPassthrough = value == "true"
+		hasCORS = true
+	}
+	if !hasCORS {
+		return nil
```

Comment:
How does `cors.New` handle a `cors.Options` input struct with fields having zero values? 
I presume it just ignores these CORS-wise since we can already end up with only some of the supported CORS parameters being set. Thus, can't we just always return a struct value (as opposed to a pointer)?